server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/IsMarkedAsSpam.java

```diff
@@ -66,18 +65,17 @@ public void init() throws MessagingException {
 
     @Override
     public Collection<MailAddress> match(Mail mail) {
-        Serializable attribute = mail.getAttribute(SpamAssassinResult.STATUS_MAIL_ATTRIBUTE_NAME);
-        if (isMarkedAsSpam(attribute)) {
-            return mail.getRecipients();
-        }
-        return ImmutableList.of();
+        return mail.getRecipients()
+            .stream()
+            .filter(recipient -> isMarkedAsSpam(mail, recipient))
+            .collect(Guavate.toImmutableList());
     }
 
-    private boolean isMarkedAsSpam(Serializable attribute) {
-        return attribute instanceof String &&
-            ((String) attribute)
-                    .toLowerCase(Locale.US)
-                    .startsWith(YES);
+    public boolean isMarkedAsSpam(Mail mail, MailAddress recipient) {
+        return mail.getPerRecipientSpecificHeaders().getHeadersForRecipient(recipient)
+            .stream()
+            .filter(header -> header.getName().equals(SpamAssassinResult.STATUS_MAIL_ATTRIBUTE_NAME))
+            .anyMatch(header -> header.getValue().toLowerCase(Locale.US).startsWith(YES));
```

Comment:
this means we need to change the prod configuration when merging this

Reply:
Not necessarily. Did you see any configuration changes in the commits I was involved in?
(IsMarkedAsSpam and SPamAssassin are both changing, hence no harm)